CONTRIBUTING.md

```diff
@@ -0,0 +1,103 @@
+# Contributing to HClib
+
+Before opening a pull request,
+please be sure to read through these guidelines.
+Following the guidelines listed below will make it much more likely
+for your contributions to be quickly accepted and merged.
+
+## Licensing
+
+All contributions submitted to this project are governed
+by the terms outlined in our license agreement.
+Specifically, our license (Apache 2.0)
+has an explicit contributions clause in [section
+5](https://github.com/habanero-rice/hclib/blob/6ad778a/LICENSE#L131-L137):
+
+> Unless You explicitly state otherwise, any Contribution intentionally
+> submitted for inclusion in the Work by You to the Licensor shall be under
+> the terms and conditions of this License, without any additional terms or
+> conditions.  Notwithstanding the above, nothing herein shall supersede or
+> modify the terms of any separate license agreement you may have executed
+> with Licensor regarding such Contributions.
+
+## Commits
+
+Please follow established best practices for your commits.
+Do only one thing per commit, and write a descriptive commit message.
+[Chris Beams' *seven rules of a great Git commit
+message*](https://chris.beams.io/posts/git-commit/#seven-rules)
+provides a good set of guidelines:
+
+1. Separate subject from body with a blank line
```

Comment:
Could you show me an example of the type of commit you're thinking of? For example, at the moment I don't believe a single commit in this repo has both a subject and a body, so that would be a pretty disruptive change to everyone's development practices today. That doesn't mean it's a bad idea, just something that's definitely going to need more discussion. In general, most repos don't have subject and body, but if it's something you feel strongly about we can definitely talk about adding that requirement.